maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gittest/src/main/java/org/apache/maven/scm/provider/git/command/checkout/GitSshCheckOutCommandTckTest.java

```diff
@@ -0,0 +1,190 @@
+
+    protected GitSshCheckOutCommandTckTest() throws GeneralSecurityException
+    {
+         sshServer = CoreTestSupportUtils.setupTestServer( getClass() );
+         keyPair = CommonTestSupportUtils.generateKeyPair( KeyUtils.RSA_ALGORITHM, 2048 );
+         acceptedPublicKeys = new ArrayList<>();
+         acceptedPublicKeys.add( keyPair.getPublic() );
+         PublickeyAuthenticator authenticator = new KeySetPublickeyAuthenticator( "onlykey",
+                  acceptedPublicKeys );
+         sshServer.setPublickeyAuthenticator( authenticator );
+    }
+
+    void writePrivateKeyAsPkcs8( Path file, String passphrase )
+            throws IOException, GeneralSecurityException
+    {
+        // encryption only optional
+        if ( passphrase != null )
+        {
+            // encryption with format outlined in https://dnaeon.github.io/openssh-private-key-binary-format/
+            OpenSSHKeyPairResourceWriter writer = new OpenSSHKeyPairResourceWriter();
+            OpenSSHKeyEncryptionContext context = new OpenSSHKeyEncryptionContext();
+            context.setCipherType( "192" );
+            context.setPassword( passphrase );
+            try ( OutputStream output = Files.newOutputStream( file ) )
+            {
+                writer.writePrivateKey( keyPair, "comment", context, output );
+            }
+        }
+        else 
+        {
+            // wrap unencrypted private key as regular PKCS8 private key
+            PKCS8Generator pkcs8Generator = new JcaPKCS8Generator( keyPair.getPrivate(), null );
+            PemObject pemObject = pkcs8Generator.generate();
+            
+            try ( Writer writer = Files.newBufferedWriter( file );
+                            JcaPEMWriter pw = new JcaPEMWriter( writer ) )
+            {
+                pw.writeObject( pemObject );
+            }
+        }
+
+        if ( file.getFileSystem().supportedFileAttributeViews().contains( "posix" ) )
+        {
+            // must only be readable/writeable by me
+            Files.setPosixFilePermissions( file, PosixFilePermissions.fromString( "rwx------" ) );
+        }
+    }
+
+    protected abstract String getScmProvider();
+
+    /** {@inheritDoc} */
+    public String getScmUrl()
+        throws Exception
+    {
+        return "scm:" + getScmProvider() + ":ssh://localhost:" + sshServer.getPort() + "/repository";
+    }
+
+    public void configureCredentials( ScmRepository repository, String passphrase ) 
+        throws Exception
+    {
+        ScmProviderRepositoryWithHost providerRepository = 
+             ScmProviderRepositoryWithHost.class.cast( repository.getProviderRepository() );
+        // store as file
+        Path privateKeyFile = tmpDirectory.newFile().toPath();
+        writePrivateKeyAsPkcs8( privateKeyFile, passphrase );
+        providerRepository.setPrivateKey( privateKeyFile.toString() );
+        if ( passphrase != null )
+        {
+            providerRepository.setPassphrase( passphrase );
```

Comment:
Why test, set it right away, if it is internally, `null` then the result will be the same, no?

Reply:
Done in https://github.com/apache/maven-scm/pull/155/commits/f03834594a6c6057886701b78156cd08333711eb.